### PR TITLE
adding $allowedxmlnamedentities to global

### PR DESCRIPTION
**Pull request by ovidiul**
This PR should address a naming mismatch and make the `$allowedxmlnamedentities` referenced on the kses.php file a global variable

Trac ticket: https://core.trac.wordpress.org/ticket/54060

### CODE REVIEW

File: tests/phpunit/tests/kses.php [hunk 1]
```diff
@@ -7,6 +7,51 @@
  */
 class Tests_Kses extends WP_UnitTestCase {
 
+	/**
+	 * Test callback for `wp_kses_normalize_entities()` regular expression.
+	 *
+	 * @dataProvider  data_wp_kses_xml_named_entities
+	 *
+	 * @ticket 54060
```
Comment (peterwilsoncc) [inline tests/phpunit/tests/kses.php:15]:
```suggestion
	 * @dataProvider  data_wp_kses_xml_named_entities
	 * @covers ::wp_kses_xml_named_entities
	 * @ticket 54060
```

**Review by peterwilsoncc — COMMENTED** [review]
Looks good to me with one minor change to the docblock in the tests.

I've edited the PR to target `trunk` as the default branch has changed since this PR was opened.

File: tests/phpunit/tests/kses.php (hunk 1 above)
Comment (ovidiul) [reply tests/phpunit/tests/kses.php]: Hi @peterwilsoncc, thanks for the suggestion, annotation is added now.

**Review by costdev — COMMENTED** [review]
Hi @ovidiul! I've sent you a message on Slack with a `diff` with suggestions for this PR.

**Comment by ovidiul** [comment]
> Hi @ovidiul! I've sent you a message on Slack with a `diff` with suggestions for this PR.

Thank you @costdev but I don't think I've received your message yet? Feel free to post the diff here as well.

**Comment by costdev** [comment]
@ovidiul No problem!

Here's [the diff](https://github.com/WordPress/wordpress-develop/files/7568837/54060.1.zip) and the details:

- [x] **Tidies up the docblock a little bit.**
For consistency with docblock structure in other tests in the test suite + Clarity about the purpose of each argument.

- [x] **Adds named keys to the datasets.**
For readability for other tests (and our future selves).

- [x] **Replaces the `html_entity_decode()` calls with the decoded characters in the data provider.**
When testing a function, we should not use any functionality used directly in the function itself.

- [x] **Replaces `html_entity_decode( '&nbsp;' )` with `utf8_encode( chr( 160 ) )`.**
When testing a function, we should not use any functionality used directly in the function itself.
`&nbsp;` renders ASCII 160 rather than ASCII 32. See [here](https://www.php.net/manual/en/function.html-entity-decode.php#Notes).

- [x] **Adds a test and data provider to check that the globals that `kses.php` should define, are defined.**
If this is needed.

**Comment by ovidiul** [comment]
Thank you for the changes 🙇 @costdev they look great, I've merged them to this PR.

**Comment by SergeyBiryukov** [comment]
Thanks for the PR! This looks good to me, I only have two minor notes:
* For consistency with the existing `$allowedentitynames` variable, I think it would be better to standardize on `$allowedxmlentitynames` here, instead of `$allowedxmlnamedentities`.
* The `null` array key should probably be `empty string`, to avoid confusion with an actual `null` value, which is not being tested here.

I can make these adjustments on commit.

**Comment by SergeyBiryukov** [comment]
Merged in https://core.trac.wordpress.org/changeset/52229.

**Pull request closed without merging**